### PR TITLE
feat: alternativepath support

### PR DESCRIPTION
**Pull request by jeswr**
Thanks for this project!

I started something similar years ago but never saw it to completion https://github.com/jeswr/shacl-query; might be able to extract some bits of it though to cover more types of shapes.

This PR hackily adds support for `alternativePath`s which is something that I needed support for on something I was doing today.

I'm happy to clean this up if you can guide me on how to correctly extract the values pattern in the `shapeToPattern`s file.

### CODE REVIEW

**Comment by tpluscode** [comment]
Hey. Thank you so much for this. The internal API is not yet finalised. I have recently been thinking about extensibility, such as that constraints, including custom shape properties can be plugged in. Just FYI

Let me have a look at your PR now. It is a draft because you're still working on it or simply so that I don't merge yet?

File: packages/shape-to-query/package.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "name": "@hydrofoil/shape-to-query",
   "version": "0.3.2",
   "scripts": {
-    "prepack": "tsc"
+    "prepack": "tsc",
+    "prepare": "npm run prepack"
```
Comment (tpluscode) [inline packages/shape-to-query/package.json:6]: Curious, why did you add `prepare`?

**Review by tpluscode — COMMENTED** [review]
I see how this is still a WIP. The only fundamental thing will be to support alternative of any arbitrary length

File: packages/shape-to-query/package.json (hunk 1 above)
Comment (jeswr) [reply packages/shape-to-query/package.json:6]: This is just so I can locally install it for what I'm working on at the moment; will remove when I take this out of draft.

Comment (tpluscode) [reply packages/shape-to-query/package.json:6]: gotcha

File: packages/shape-to-query/lib/shapeToPatterns.ts [hunk 2]
```diff
@@ -129,6 +138,25 @@ function * deepPropertyShapePatterns({ shape, focusNode, options, parentPatterns
...
+
+      if (altList === null) {
+        continue
+      }
+
+      const property = [...altList].map(e => e.term)
+
+      if (!property.every(term => term.termType === 'NamedNode')) {
+        continue
+      }
```
[the remark this answers is not in the document]
Comment (jeswr) [reply packages/shape-to-query/lib/shapeToPatterns.ts:152]: IIRC I was on the way to supporting arbitrary SHACL paths [at least this code snippet would indicate so anyway](https://github.com/jeswr/shacl-query/blob/12f2273c9a58d82fdf84b84c9103964aafc28bdf/lib/basic-extract.ts#L113-L140). 

I'll revisit this on the weekend and try and recall the algorithm that was being used to achieve this.

[the remark this answers is not in the document]
Comment (tpluscode) [reply packages/shape-to-query/lib/shapeToPatterns.ts:152]: One step at a time. Maybe I'll create an issue to discuss how the end result could look like for some paths

[the remark this answers is not in the document]
Comment (tpluscode) [reply packages/shape-to-query/lib/shapeToPatterns.ts:152]: Hey, I realise that I think I do already have what is needed: https://github.com/hypermedia-app/clownface-shacl-path#tosparqlsequence

I need to see if the `toSequence` function works nicely will more complex functions but I pretty much added it to support this scenario, where a SPARQL path needs to be deconstructed in order to capture all stages as variables...
